atum/src/test/scala/za/co/absa/atum/utils/controlmeasure/ControlMeasureBuilderTest.scala

```diff
@@ -65,7 +65,7 @@ class ControlMeasureBuilderTest extends AnyFlatSpec with SparkTestBase with Matc
       ControlMeasureMetadata("SourceApp1", "CZ", "HistoryType1", "input/path1", "SourceType1", 1, testingDate, Map()),
       None,
       List(
-        Checkpoint("InitCheckpoint1", Some("Atum"), Some(testingVersion), testingDateTime1, testingDateTime2, "Workflow1", 1, List(
+        Checkpoint("InitCheckpoint1", Some(testingSoftware), Some(testingVersion), testingDateTime1, testingDateTime2, "Workflow1", 1, List(
```

Comment:
Here, I find it hard to read what the expected `software` string will be. It's loaded from a different file (probably my mistake, as you only followed suit of `testingVersion`). I have realized this specifically in case one wanted to show the alternation of the `software` and `version` values loaded from config.

So just if you happened to be digging around this, this is just something I have realized is seemingly suboptimal. But if it would be too much of a detour, forget it.

Reply:
So I tried something. Please review if this is how you imagined it.

- I have moved the "constants" to a `BasteTestSuite` where they are taken by classes that need it while the `ControlInfoToJsonSerializationSpec` that uses plane BuildProperties doesn't use it at all. The constants now also have a prefix of `BasteTestSuite`
- I have created some small tests for "default" `BuildProperties` in Atum. Saying they need to adhere to the name `atum_scalaVersion` and that the version needs to be `semVer` parsable.

The next point I would like to discuss is about the "new BuildProperties". Here I did not do anything as I realized:

- This PR is actually not implementing some new feature. It is just adding App name to App version that was already there
- By changing the BuildProperties, even if for the better, it will change the API and the change will touch more places. This deserves a proper PR and will also have to be with a new Major version. 

So I propose to cut a branch from master called release-4.0.0 and start slowly working on "API breaking" stuff there and when we accumulate enough or find it is time, we release.
I will be happy to do that if we agree.

Reply:
> I have created some small tests for "default" BuildProperties in Atum. Saying they need to adhere to the name atum_scalaVersion and that the version needs to be semVer parsable.

That's very nice. I have created a suggestion branch that extends this on some of the tests, too: https://github.com/AbsaOSS/atum/tree/feature/79-software-and-version-from-build-suggestions1

> So I propose to cut a branch from master called release-4.0.0 and start slowly working on "API breaking"

That makes sense, too.